miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -1135,6 +946,10 @@ export.exec_kernel_proc
     mem_loadw
     # => [KERNEL_PROCEDURE_HASH, <procedure_inputs>, <pad>]
 
+    # store the KERNEL_PROCEDURE_HASH to the memory to be able to call it dynamically
+    exec.memory::store_dyn_proc_root
+    # => [kernel_procedure_hash_addr, <procedure_inputs>, <pad>]
```

Comment:
Related to some other comments, this seems to be redundant as we first load procedure hash from memory and then store it again.

Also, the comments for this procedure need to be updated as we are no longer limited by 12 elements, I believe.